From 8dd54d0a3b3657730518d184962939582c593378 Mon Sep 17 00:00:00 2001
From: Zhou Peng <eagle.zhou@nxp.com>
Date: Tue, 24 Jul 2018 09:48:25 +0800
Subject: [PATCH 4229/5242] - MLK-19004 - [i.MX8QXP/VPU]: Add self-defined EOS
 tag instead of standard tag

commit  d98482678ea6da6d7026f58daacca4c69497a40b from
https://source.codeaurora.org/external/imx/linux-imx.git

Add self-defined EOS tag for some formats, includes: Mpeg2 and Mjpeg

Signed-off-by: Zhou Peng <eagle.zhou@nxp.com>
(cherry picked from commit 6550b3dca4eaf6c42c38234a6e8bbf09072d9f54)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c |   14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)
 mode change 100644 => 100755 drivers/mxc/vpu-decoder-b0/vpu_b0.c

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
old mode 100644
new mode 100755
index f8df690a..d65760c
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -50,6 +50,11 @@
 
 unsigned int vpu_dbg_level_decoder = 1;
 
+/* Generic End of content startcodes to differentiate from those naturally in the stream/file */
+#define EOS_GENERIC_HEVC 0x7c010000
+#define EOS_GENERIC_JPEG 0xefff0000
+#define EOS_GENERIC_MPEG 0xCC010000
+
 static void vpu_api_event_handler(struct vpu_ctx *ctx, u_int32 uStrIdx, u_int32 uEvent, u_int32 *event_data);
 static void v4l2_vpu_send_cmd(struct vpu_ctx *ctx, uint32_t idx, uint32_t cmdid, uint32_t cmdnum, uint32_t *local_cmddata);
 static bool add_eos(struct vpu_ctx *ctx, u_int32 uStrBufIdx);
@@ -896,8 +901,7 @@ static int v4l2_ioctl_streamoff(struct file *file,
 	} else if (ctx->firmware_stopped) {
 		vpu_dbg(LVL_ERR, "%s(): not succeed and firmware is stopped\n", __func__);
 		return -EINVAL;
-	}
-	else
+	} else
 		return ret;
 }
 
@@ -1095,7 +1099,8 @@ static bool add_eos(struct vpu_ctx *ctx, u_int32 uStrBufIdx)
 		last = 0x0a010000;
 		break;
 	case VPU_VIDEO_MPEG2:
-		last = 0xb7010000;
+		//last = 0xb7010000;
+		last = EOS_GENERIC_MPEG;
 		break;
 	case VPU_VIDEO_ASP:
 		last = 0xb1010000;
@@ -1107,6 +1112,9 @@ static bool add_eos(struct vpu_ctx *ctx, u_int32 uStrBufIdx)
 	case VPU_VIDEO_RV9:
 		last = 0x34010000;
 		break;
+	case VPU_VIDEO_JPEG:
+		last = EOS_GENERIC_JPEG;
+		break;
 	case VPU_VIDEO_HEVC:
 		last = 0x4A010000;
 		last2 = 0x20;
-- 
1.7.9.5

